### ISSUE TITLE
Exposed function defintions

### ISSUE THREAD

**Issue by Noitidart** [Noitidart/FoxSpeak#1]
Hey @canaar can you please list me the exposed functions from the .so file. Like `function blah(uint, bool)` so i can then start writing some ctypes to use it. :)

**Comment by 55abhilash** [comment]
I am working. By the way , just out of curiosity, in what way will the custom data types will be handled? I mean for eg. a function like : 
static int acmod_init_feat(acmod_t *acmod);
How will acmod_t be used in ctypes?

**Comment by Noitidart** [comment]
Thanks its possible to handle custom data types I just have to see the raw source to see what it was defined as. Multiple methods, including option of just pointer to it, `ctypes.void_t.ptr` or an opaque structure pointer.

**Comment by 55abhilash** [comment]
Fine. I am on it.

**Comment by Noitidart** [comment]
Thanks no rush but if you can upload the raw source file too to the .so that would be awesome. I'm busy with Profilist right now but I can start doing some work on this too :)

**Comment by 55abhilash** [comment]
Added the list of function declarations , libpocketsphinx_fns.txt, to modules. Listed them using this script : https://github.com/canaar/ListFns . .Let me know in case of problems.

**Comment by Noitidart** [comment]
Fantastic Ill look it over and try to hook up some ctypes. I gotta go buy a mic today to test this haha

**Comment by Noitidart** [comment]
Hey man can you please share with me the header files I need to see it to define the types. `sphinxbase/prim_type.h` looks like an important one.

**Comment by 55abhilash** [comment]
Yeah wait I'll share all headers.

**Comment by Noitidart** [comment]
Fantastic work thanks @canaar I'm going to be pretty busy for the next couple weeks but please continue to upload the dll and the dylib for windows and mac. I'll make sure to do the ctypes work so you can see it in action.

**Comment by 55abhilash** [comment]
Yeah, dll and dylib is gonna take some time coz ive never worked in win or mac environment. Will let you know when I proceed...

**Comment by Noitidart** [comment]
No problem you're doing fantastic!

**Comment by yajd** [comment]
This is an amazing piece of work, I am eagerly looking to see this completed. When do you guys think you will be done?

**Comment by 55abhilash** [comment]
Sorry I am taking too long. Was busy for a few weeks. I will dedicate more time to this now.

**Comment by Noitidart** [comment]
Nooo problem I'm actually doing lots of practice for this addon. I am so excited about getting the Windows and OSX libraries from your end. I am practicing by making and relasing a native screenshot addon this week! https://github.com/Noitidart/NativeShot

It is giving me so much practice with js-ctypes and C libraries.

I am very very glad to hear from you and you're still motivated about it I got worried that you lost interest. I am so glad! :)

**Comment by 55abhilash** [comment]
@yajd This will take some time. @Noitidart is unable to do C. I know C, but am a noob with no big development experience. So, it'll take time. ;) Cheers

**Comment by Noitidart** [comment]
Ooops missed your post @yajd but @canaar  is right on the dot. Him and I are collaborating and will get it out though. We won't let you down.

**Comment by 55abhilash** [comment]
@Noitidart  Consider the windows .dll to be done today by evening Indian standard time.

**Comment by Noitidart** [comment]
Thanks @canaar so much! I actually got carried away with my NativeShot addon but as soon as the windows one is ready ill do some work on here as doing on windows is easiest for me as thats my default os. My linux and osx are on VMs which is a headache haha
